packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1183,6 +1195,10 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         ReportOpPerfTelemetry(this.context.clientId, this.deltaManager, this.logger);
     }
 
+    private setReadOnly(): void {
+        (this.context as any).setReadOnly();
```

Comment:
This is never good idea - you need to add API to definitions and propagate them correctly before you can make these changes. Fine for prototyping.